File: frontend/src/features/notifications/types/notification.ts
export type AppNotification = {
  id: number
  event_type: string
  title: string
  message: string
  student_id: number | null
  resource_type: string | null
  resource_id: number | null
  action_url: string | null
  read_at: string | null
  created_at: string
}

export type NotificationList = {
  items: AppNotification[]
  unread_count: number
}

File: backend/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationResponse(BaseModel):
    id: int
    event_type: str
    title: str
    message: str
    student_id: int | None
    resource_type: str | None
    resource_id: int | None
    action_url: str | None
    read_at: datetime | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    unread_count: int


class NotificationUnreadCount(BaseModel):
    unread_count: int
